dashboard/src/components/Catalog/CatalogItem.tsx

```diff
@@ -38,32 +42,55 @@ function trimDescription(desc: string): string {
 }
 
 const CatalogItem: React.SFC<ICatalogItemProps> = props => {
-  const { item } = props;
-  const { icon, name, repo, version, description, type, namespace, id, csv } = item;
-  const iconSrc = icon || placeholder;
-  let link;
-  let tag1;
-  let subIcon;
-  if (type === "chart") {
-    tag1 = (
-      // TODO(#1803): See #1803 regarding the work-arounds below for the fact
-      // that repo is required if we're dealing with a chart here.
-      <Link
-        className="ListItem__content__info_tag_link"
-        to={url.app.repo(repo?.name || "", namespace)}
-      >
-        {repo?.name}
-      </Link>
-    );
-    link = url.app.charts.get(name, repo || {} as IRepo, namespace);
-    subIcon = helmIcon;
+  if (props.type === "operator") {
+    const item = props.item as IOperatorCatalogItem;
+    return OperatorCatalogItem(item); 
   } else {
-    // Cosmetic change, remove the version from the csv name
-    const csvName = csv?.split(".v")[0];
-    tag1 = <span>{csvName}</span>;
-    link = `/ns/${namespace}/operators-instances/new/${csv}/${id}`;
-    subIcon = operatorIcon;
+    const item = props.item as IChartCatalogItem;
+    return ChartCatalogItem(item);
   }
+};
+
+const OperatorCatalogItem: React.SFC<IOperatorCatalogItem> = props => {
+  const { icon, name, csv, version, description, namespace, id } = props;
+  const iconSrc = icon || placeholder;
+  // Cosmetic change, remove the version from the csv name
+  const csvName = props.csv.split(".v")[0];
+  const tag1 = <span>{csvName}</span>;
+  const link = `/ns/${namespace}/operators-instances/new/${csv}/${id}`;
```

Comment:
this link should be obtained from a function :)

Reply:
Yeah - grepping for `"/ns/.*operator"` shows quite a few operator URLs in non-test code that aren't using the shared urls helper (or even have a url helper defined as in this case for the operator-instance new).